### PR TITLE
feat:request magic link

### PR DESCRIPTION
**Pull request by mariams58**
​

## Description

This PR implements the following functionality:

**Magic Link Creation Endpoint**
+ A new endpoint `api/v1/auth/request-magic-link` is implemented to generate a unique link
+ This endpoint uses a `send_mail` service to send the unique link to the user's email address

## Related Issue (Link to issue ticket)

#476 - {FEAT] - Request Magic Link
​
### File Changes

1. **api/v1/routes/auth.py:**
+ Implemented the `request-magic-link` endpoint
+ Validate the email provided to know if user is valid
2. **api/v1/schemas/user.py:**
+ Create schema `MagicLinkRequest` to handle the request data
3. **tests/v1/test_magic_link.py:**
+ Created a test suite for the `request-magic-link` endpoint
+ Used mock services and test-specific configurations to ensure isolated and accurate testing

​## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Rename your PR to 
feat: request....

File: api/utils/send_mail.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    receiver_email = email
+    password = settings.MAIL_PASSWORD
+
+    message = MIMEMultipart("alternative")
+    message["Subject"] = "Your Magic Link"
+    message["From"] = sender_email
+    message["To"] = receiver_email
+
+    text = f"Use the following link to log in: http://localhost:8000/magic-link?token={token}"
+    html = f"<html><body><p>Use the following link to log in: <a href='http://localhost:8000/magic-link?token={token}'>Magic Link</a></p></body></html>"
```
Comment (joboy-dev) [inline api/utils/send_mail.py:19]: That hostname should be dynamic
You don't want a user to see local host in their email

So update line 18 andb19

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

File: tests/v1/user_deactivation_test.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 client = TestClient(app)
 DEACTIVATION_ENDPOINT = '/api/v1/users/deactivation'
 LOGIN_ENDPOINT = 'api/v1/auth/login'
+MAGIC_ENDPOINT = '/api/v1/auth/request-magic-link'
```
Comment (joboy-dev) [inline tests/v1/user_deactivation_test.py:22]: please remove this

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

**Review by Takinnuoye5 — APPROVED** [review]
approve

**Pull request merged**